src/rdkafka_transport.c

```diff
@@ -920,6 +920,61 @@ int rd_kafka_transport_ssl_ctx_init (rd_kafka_t *rk,
                 }
 	}
 
+	/* Andrea Minuto (PKCS12) */
+	if (rk->rk_conf.ssl.pkcs12.keystore_location) {
+		FILE *fp;
+		EVP_PKEY *pkey;
+		X509 *cert;
+		STACK_OF(X509) *ca = NULL;
+		PKCS12 *p12;
+
+		rd_kafka_dbg(rk, SECURITY, "SSL",
+		"Loading client's keystore file from %s",
+		rk->rk_conf.ssl.pkcs12.keystore_location);
+
+		if (!rk->rk_conf.ssl.pkcs12.keystore_password) {
+			rd_snprintf(errstr, errstr_size,
+				"The ssl.pkcs12.keystore.password is mandatory: ");
+			goto fail;
+		}
+
+		ERR_load_crypto_strings();
+		if (!(fp = fopen(rk->rk_conf.ssl.pkcs12.keystore_location, "rb"))) {
+			rd_snprintf(errstr, errstr_size,
+				"ssl.pkcs12.keystore.location failed: ");
```

Comment:
`"Failed to open ssl.pkcs12.keystore.location: %s: %s", ..key_store.location, rd_strerror(rd_errno());`